BetterStats/BetterStats.cs

```diff
@@ -755,15 +760,37 @@ public static void AddPlanetFactoryData(PlanetFactory planetFactory)
                 {
                     continue;
                 }
-                if (generator.productId == 0 || generator.productHeat == 0)
+                var isFuelConsumer = generator.fuelHeat > 0 && generator.fuelId > 0 && generator.productId == 0;
+                if ((generator.productId == 0 || generator.productHeat == 0) && !isFuelConsumer)
                 {
                     continue;
                 }
-                var productId = generator.productId;
-                EnsureId(ref counter, productId);
 
-                counter[productId].production += 60.0f * TICKS_PER_SEC * generator.capacityCurrentTick / generator.productHeat;
-                counter[productId].producers++;
+                if (isFuelConsumer)
+                {
+                    // account for fuel consumption by power generator
+                    var productId = generator.fuelId;
+                    EnsureId(ref counter, productId);
+
+                    counter[productId].consumption += 60.0f * TICKS_PER_SEC * generator.useFuelPerTick / generator.fuelHeat;
+                    counter[productId].consumers++;
+                }
+                else
+                {
+                    var productId = generator.productId;
+                    EnsureId(ref counter, productId);
+
+                    counter[productId].production += 60.0f * TICKS_PER_SEC * generator.capacityCurrentTick / generator.productHeat;
+                    counter[productId].producers++;
+                    if (generator.catalystId > 0)
+                    {
+                        // account for production of critical photons by ray receivers. graviton lenses are consumed at a rate of 0.1/min
+                        // TODO: figure out where the game stores this rate
+                        EnsureId(ref counter, generator.catalystId);
+                        counter[generator.catalystId].consumption += RAY_RECEIVER_GRAVITON_LENS_CONSUMPTION_RATE_PER_MIN;
```

Comment:
I have this value as a constant right now because I could not find out where the game actually gets it from. I assume it's not hard-coded because it has changed before.

Reply:
It doesn't seem like there is an easily accessible way to get this value. The UI has the "0.100 / min" text hard coded and the logic to decrement `catalystPoint` is spread between `PowerGeneratorComponent.GameTick_Gamma()` and `PowerSystem.GameTick()` (thanks to  Selsion on discord). We _could_ get this by observing the change in the `catalystPoint` for one particular generator between ticks, but that seems to be somewhat error prone. So, for now, I just kept it as a constant